pkg/landscaper/blueprints/resolve.go

```diff
@@ -80,60 +73,75 @@ func Resolve(ctx context.Context, resolver ctf.ComponentResolver, cdRef *lsv1alp
 		return nil, fmt.Errorf("unable to resolve component descriptor for ref %#v: %w", cdRef, err)
 	}
 
-	blueprint, err := ResolveBlueprintFromBlobResolver(ctx, cd, blobResolver, fs, bpDef.Reference.ResourceName)
-	if err != nil {
-		return nil, err
-	}
-	return New(blueprint, readonlyfs.New(fs)), nil
+	return ResolveBlueprintFromBlobResolver(ctx, cd, blobResolver, bpDef.Reference.ResourceName)
 }
 
 // ResolveBlueprintFromBlobResolver resolves a blueprint defined by a component descriptor with
 // a blob resolver.
-func ResolveBlueprintFromBlobResolver(ctx context.Context,
+func ResolveBlueprintFromBlobResolver(
+	ctx context.Context,
 	cd *cdv2.ComponentDescriptor,
 	blobResolver ctf.BlobResolver,
-	fs vfs.FileSystem,
-	blueprintName string) (*lsv1alpha1.Blueprint, error) {
+	blueprintName string) (*Blueprint, error) {
 
 	// get blueprint resource from component descriptor
 	resource, err := GetBlueprintResourceFromComponentDescriptor(cd, blueprintName)
 	if err != nil {
 		return nil, err
 	}
 
-	var data bytes.Buffer
-	// todo: do not store the blueprint in memory and rather directly decode the bytestream to the filesystem.
-	blobInfo, err := blobResolver.Resolve(ctx, resource, &data)
+	if blueprint, err := GetStore().Get(ctx, cd, resource); err == nil {
+		return blueprint, nil
+	}
+	// blueprint was not cached so we need to fetch the blueprint blob and store it in the cache
```

Comment:
Don't we need to check the error returned by `GetStore().Get(ctx, cd, resource)` like

`if error != NotFoundError {
  return nil, err
}`

Reply:
I explicitly did not handled that error as I'm was unsure what to do about such errors.
In this case errors could happen due to corrupted blueprints or temporary filesystem failures.
In both I wanted the application to give a recovery chance by re-downloading the blueprint and restoring it in the store.

Both cases could have negative side effect:
- either we are in a infinite reconcile loop due to a corrupted disk/corrupted blueprint with no chance to recover
- or we are downloading the blueprint to often (but still would get the error handled in line 133 when the blueprint is tried to be stored)

WDYT how should we handle such issues?